### ISSUE TITLE
[3.0 dev] `phpdoc_types_order` does not work together with array shapes

### ISSUE THREAD

**Issue by ruudk** [FriendsOfPHP/PHP-CS-Fixer#5798]
### Code snippet that reproduces the problem

```php
    /**
     * @return array<array{level:string, message:string, context:array<mixed>}>
     */
    public function getLogs(?string $level = null) : array
    {
    }
```

```
PHP CS Fixer 3.0.1-DEV Constitution by Fabien Potencier and Dariusz Ruminski
Runtime: PHP 8.0.7
Loaded config default from "/Volumes/CS/www/website/.php-cs-fixer.php".
Using cache file "var/.php_cs.cache".
Paths from configuration file have been overridden by paths provided as command arguments.
F                                                                                                                            1 / 1 (100%)
Legend: ?-unknown, I-invalid file syntax (file ignored), S-skipped (cached or empty file), .-no changes, F-fixed, E-error
   1) /Volumes/CS/www/website/src-dev/Testing/Logging/InMemorLogger.php (phpdoc_types_order)
      ---------- begin diff ----------
--- /Volumes/CS/www/website/src-dev/Testing/Logging/InMemorLogger.php
+++ /Volumes/CS/www/website/src-dev/Testing/Logging/InMemorLogger.php
@@ -97,7 +97,7 @@
     }

     /**
-     * @return array<array{level:string, message:string, context:array<mixed>}>
+     * @return array<array{level:string, message:string, context:array<mixed>|}>
      */
     public function getLogs(?string $level = null) : array
     {

      ----------- end diff -----------


Checked all files in 0.085 seconds, 18.000 MB memory used
exit status 8
```

**Comment by kubawerlos** [comment]
Duplicate of https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/5460

**Comment by ruudk** [comment]
It might seem so, but I'm using `3.0` branch. So even if it is a duplicate, it still doesn't work?

**Comment by ruudk** [comment]
@kubawerlos https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/5395 seems to be merged to 2.9 but that is also merged into 3.0 , right?

**Comment by ruudk** [comment]
It's def not a duplicate, as my issue is not a removal of `}` at the end, but a prefix of `|` before `}`

See diff:
```diff
-     * @return array<array{level:string, message:string, context:array<mixed>}>
+     * @return array<array{level:string, message:string, context:array<mixed>|}>
```

**Comment by julienfalque** [comment]
I also think this is a different issue, I'd like to investigate it to be sure.

**Comment by ruudk** [comment]
@keradus FYI, Just tried the newly released 3.0.1 and it does contain this bug. It means that people upgrading from 3.0.0 to 3.0.1 will experience this problem when they are using array shapes.

**Comment by Gemorroj** [comment]
my workaround 😂
```
"friendsofphp/php-cs-fixer": "^3.0, !=3.0.1"
```
